src/wh_server_crypto.c

```diff
@@ -296,30 +302,47 @@ static int hsmCryptoCurve25519(whServerContext* server, whPacket* packet,
 #ifdef HAVE_ECC
 static int hsmCacheKeyEcc(whServerContext* server, ecc_key* key, whKeyId* outId)
 {
+    uint8_t* cacheBuf;
+    whNvmMetadata* cacheMeta;
     int ret;
-    int slotIdx = 0;
-    uint32_t qxLen;
-    uint32_t qyLen;
-    uint32_t qdLen;
+    uint32_t qxLen = 0;
+    uint32_t qyLen = 0;
+    uint32_t qdLen = 0;
     whKeyId keyId = WOLFHSM_KEYTYPE_CRYPTO;
+    /* get the lengths since they vary by ecc size, expect BUFFER_E since
+     * LENGTH_ONLY_E isn't supported. Note that this is a dangerous operation 
+     * but by having size of 0 or buffer of NULL we avoid NULL checks when
+     * getting the length */
```

Comment:
looks like those args are NULL checked in that wolfCrypt function, so shouldn't necessarily be dangerous - can you elaborate on what you mean here?

Reply:
It's dangerous because I have to pass in a non-NULL nonsense pointer for the length I'm trying to get, the operation will fail as long as at least one of the lengths, qx or qy, is NULL so it should be good but it warrants a comment, might be wise a refactor ecc to allow a `LENGTH_ONLY_E` return